### ISSUE TITLE
Create single worker node cluster failing on 2.5.1

### ISSUE THREAD

**Issue by mbelchin** [vmware/container-service-extension#471]
Hi,

I'm trying to create a single worker node cluster as I did in the past using this command:

```
vcd cse cluster create my-cluster --network network.internal --nodes 0 --ssh-key ~/.ssh/id_rsa.pub --memory 2048 --cpu 2 --template ubuntu-16.04 --storage-profile Basic
```

Now I'm trying to create it using: 

```
vcd cse cluster create my-cluster --network network.internal --nodes 0 --ssh-key ~/.ssh/id_rsa.pub --memory 2048 --cpu 2 --storage-profile Basic --template-name ubuntu-16.04_k8-1.15_weave-2.5.2 --template-revision 1
```

And I'm getting this error:

`Error: Worker node count must be > 0 (received 0).`

In the past I was able to create a single worker node cluster. In the documentation you can still read that as well:
https://vmware.github.io/container-service-extension/CLUSTER_MANAGEMENT.html#example

```
# create a single worker node cluster, connected to the specified network. Nodes can be added later
> vcd cse cluster create mycluster --network intranet --nodes 0 --ssh-key ~/.ssh/id_rsa.pub
````

These are my current versions:

```
$ vcd version
vcd-cli, VMware vCloud Director Command Line Interface, 22.0.0

$ vcd cse version
CSE, Container Service Extension for VMware vCloud Director, version 2.5.1

$ vcd cse template list
name                                revision  is_default    catalog    catalog_item                           description
--------------------------------  ----------  ------------  ---------  -------------------------------------  ----------------------------------------------------------------
photon-v2_k8-1.12_weave-2.3.0              1  False         cse        photon-v2_k8-1.12_weave-2.3.0_rev1     PhotonOS v2, Docker-ce 18.06.2-3, Kubernetes 1.12.7, weave 2.3.0
ubuntu-16.04_k8-1.13_weave-2.3.0           1  True          cse        ubuntu-16.04_k8-1.13_weave-2.3.0_rev1  Ubuntu 16.04, Docker-ce 18.09.7, Kubernetes 1.13.5, weave 2.3.0
ubuntu-16.04_k8-1.15_weave-2.5.2           1  False         cse        ubuntu-16.04_k8-1.15_weave-2.5.2_rev1  Ubuntu 16.04, Docker-ce 18.09.7, Kubernetes 1.15.3, weave 2.5.2
```

Any idea about the issue ?
Any workaround to get the cluster just with the master node ?

Thanks in advance.

**Comment by mbelchin** [comment]
I've tried creating the cluster with just one worker nodes and then resize it two 0 worker nodes, but that does not work either.

```
vcd cse cluster resize my-cluster --network network.internal --nodes 0
```

Any other idea ?

Thanks in advance.

**Comment by rocknes** [comment]
Hi, 

Thanks for point out the deficiency in the documentation. We will fix it.

CSE 2.5.0 and onward will not allow creation of a cluster without a dedicated worker node. The CSE cluster will always contain a master node and 1+ worker nodes. The param --nodes corresponds to *just* the worker node count in the cluster. So the smallest cluster you can create will have 1 master and 1 worker node and to get it, you will need to specify --nodes/-N 1.

Btw, I am curious, why do you want to create a cluster with only the master node?

Regards
Aritra Sen

**Comment by meltingrobot** [comment]
@rocknes It would be nice to be able to specify different specs for the master node then.  As it stands right now, I have to either size up the worker node or size down the master node after creating the cluster initially.  My worker nodes tend to be much larger than the master node in resources.

**Comment by rocknes** [comment]
Hi,

I see, there is still a way to scale down to 0 worker nodes.
vcd cse node delete <cluster name> <worker node name> lets you delete worker nodes and scale down to zero. But please make sure that no containers are running on this node. Since the deletion of the node, is not exactly graceful :(

So for your use case, the sequence of operations will look as follows
1. Deploy cluster with 1 master and 1 worker, both being small vms
2. Delete the worker node use vcd cse node delete
3. Add more nodes to the cluster (these vms being big VMs).

Let me know if this works for you.

Btw given your use case, we will look into it and see if it makes sense to relax the restriction of "at-least one worker node in the cluster".

Regards
Aritra Sen

Regards
Aritra Sen

**Comment by mbelchin** [comment]
@rocknes that is exactly what I did. Create cluster with 1 master 1 node and after that delete the worker node.

I don't really understand why it's the reason or the reasons to enforce users to create a cluster with at least 1 worker node. In the past we were able to create clusters just with the master.

Shouldn't be better to give the tools to the people and leave them to do what they really need?

As a practical case, for us is really handy to have a cluster for our developers to test features in a similar way of staging or production (some features require kubernetes features), but being low in resources. So having a cluster with just the master node, that acts as well as a worker node, is more than enough.

That help us a lot to create new clusters for new developers joining the team (we have that scripted) but also keep invoice low because we use only 1 machine with low amount of resources.

I agree with @meltingrobot and that's something I already requested in the past, be able to configure more things on cluster creation like node names templates, disk space to be used (at least some fixed values, ie: 10GB, 20GB, 40GB...) or even better be able to use LVM.
- https://github.com/vmware/container-service-extension/issues/242
- https://github.com/vmware/container-service-extension/issues/253

Looking forward to hear from you.
